ignite/cmd/plugin.go

```diff
@@ -696,6 +692,21 @@ func printPlugins(ctx context.Context, session *cliui.Session) error {
 	return nil
 }
 
+func newAppClientAPI(cmd *cobra.Command) (plugin.ClientAPI, error) {
+	// Get chain when the plugin runs inside an blockchain app
+	c, err := newChainWithHomeFlags(cmd)
+	if err != nil && !errors.Is(err, gomodule.ErrGoModNotFound) {
+		return nil, err
+	}
+
+	var options []plugin.APIOption
+	if c != nil {
+		options = append(options, plugin.WithChain(c))
```

Comment:
So the reason was that we used c without checking if it was nil or not?

Reply:
Basically yes, but is was trickier, assigning a nil `*chain.Chain` value to the `plugin.Chainer` interface when calling `plugin.WithChain()` broke a nil comparison down the line